frontend/das_haus/src/components/AuthForm.js

```diff
@@ -0,0 +1,157 @@
+              <div className="login-wrap p-4 p-md-5">
+                <div className="d-flex">
+                  <div className="w-100">
+                    <h3 className="mb-4">
+                      {isLogin ? "Sign In" : "Create a new user"}
+                    </h3>
+                  </div>
+                </div>
+                <Form method="POST" className="signin-form">
+                  <div className="form-group mb-3">
+                    <label className="label" htmlFor="name">
+                      Username
+                    </label>
+                    <input
+                      className="form-control"
+                      placeholder="Username"
+                      id="email"
+                      type="email"
+                      name="email"
+                      required
+                    />
+                  </div>
+                  <div className="form-group mb-3">
+                    <label className="label" htmlFor="password">
+                      Password
+                    </label>
+                    <input
+                      type="password"
+                      className="form-control"
+                      placeholder="Password"
+                      id="password"
+                      name="password"
+                      required
+                    />
+                  </div>
+
+                  <div className="form-group">
+                    <button
+                      type="submit"
+                      className="form-control btn btn-primary rounded submit px-3"
+                    >
+                      {isLogin ? "Sign In" : "Create a new user"}
+                    </button>
+                  </div>
+                </Form>
+                <p className="text-center">
+                  {isLogin ? "Not a member?" : "Already a member?"}{" "}
+                  <Link to={`?mode=${isLogin ? "signup" : "login"}`}>
+                    {" "}
+                    {isLogin ? "Create new user" : "Login"}
+                  </Link>
+                </p>
+              </div>
+            </div>
+          </div>
+        </div>
+      </div>
+    </section>
+  );
+}
+
+export default AuthForm;
+
+export async function action({ request, params }) {
+
+  const method = request.method;
+  const searchParams = new URL(request.url).searchParams;
+  const mode = searchParams.get("mode");
+  const data = await request.formData();
+
+  let userData = {
+    email: data.get("email"),
+    password: data.get("password"),
+  };
+
+  if (mode !== "login" && mode !== "signup") {
+    throw json({ message: "Unsupported mode." }, { status: 422 });
+  }
+
+  if (method === "POST") {
+    if (mode === "signup") {
+      let data = JSON.stringify({
+        email: userData.email,
+        password: userData.password,
+      });
+
+      let config = {
+        method: "post",
+        maxBodyLength: Infinity,
+        url: "http://localhost:8000/api/registration",
```

Comment:
don't leave the value like this. Move `http://localhost:8000` to `.env` file. Think that when for whatever reason this url will change, you will have to go back to these files and change it in every place you wrote it! It is much better to use a '.env' file for this. Then, each time the url will change, the change will be done in only one place - the .env file.

Bonus! These kind of values will almost 100% be different depending on the environment you project will be on - in most cases, you'll have a `.env` file for development, and a `.env.prod`. No need to add a `.env.prod` now, but keep this in mind!